c/src/ProtonConfig.cmake.in

```diff
@@ -22,50 +22,56 @@
 # Version: @PN_VERSION@
 # URL: http://qpid.apache.org/proton/
 
-set (Proton_VERSION       @PN_VERSION@)
+@PACKAGE_INIT@
+include("${CMAKE_CURRENT_LIST_DIR}/ProtonTargets.cmake")
 
-set (Proton_INCLUDE_DIRS  @INCLUDEDIR@)
-set (Proton_LIBRARIES     optimized @LIBDIR@/@PROTONLIB@ debug @LIBDIR@/@PROTONLIBDEBUG@)
-set (Proton_FOUND True)
+set(Proton_VERSION @PN_VERSION@)
+
+# find dependencies, because static libs don't transitively pull them
+if (Proton_USE_STATIC_LIBS)
+    set(CMAKE_MODULE_PATH_OLD ${CMAKE_MODULE_PATH})
+    set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}")
+
+    set(CyrusSASL_FOUND @CyrusSASL_FOUND@)
+    if (CyrusSASL_FOUND)
+        find_package (CyrusSASL REQUIRED)
+    endif()
+
+    set(OpenSSL_FOUND @OpenSSL_FOUND@)
+    if (OpenSSL_FOUND)
+        find_package (OpenSSL REQUIRED)
+    endif()
 
-set (Proton_Core_INCLUDE_DIRS  @INCLUDEDIR@)
-set (Proton_Core_LIBRARIES     optimized @LIBDIR@/@PROTONCORELIB@ debug @LIBDIR@/@PROTONCORELIBDEBUG@)
+    set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH_OLD})
+endif()
 
-# Add modular target in a way compatible with cmake 2.8.12
-if (NOT TARGET Proton::core)
-  add_library(Proton::core UNKNOWN IMPORTED)
-  set_target_properties(Proton::core
-    PROPERTIES
-      IMPORTED_LOCATION "@LIBDIR@/@PROTONCORELIB@"
-      IMPORTED_LOCATION_DEBUG "@LIBDIR@/@PROTONCORELIBDEBUG@"
-      INTERFACE_INCLUDE_DIRECTORIES "${Proton_Core_INCLUDE_DIRS}")
+set(Proton_INCLUDE_DIRS @INCLUDEDIR@)
+if (NOT Proton_USE_STATIC_LIBS)
+    set (Proton_LIBRARIES Proton::qpid-proton)
+else()
+    set (Proton_LIBRARIES Proton::qpid-proton-static)
 endif()
+set (Proton_FOUND True)
 
+if (NOT Proton_USE_STATIC_LIBS)
+    add_library(Proton::core ALIAS Proton::qpid-proton-core)
```

Comment:
This ALIAS thing does not work on CMake 3.11. I get

```
-- Found CyrusSASL: /usr/lib64/libsasl2.so (found version "2.1.27") 
CMake Error at /qpid-proton/build/install/lib64/cmake/Proton/ProtonConfig.cmake:83 (add_library):
  add_library cannot create ALIAS target "Proton::core" because target
  "Proton::qpid-proton-core-static" is imported but not globally visible.
```

Reply:
> New in version 3.18: An ALIAS can target a non-GLOBAL Imported Target.

So i used 3.18 feature by mistake.